website/project/views/node.py

```diff
@@ -249,7 +249,8 @@ def node_setting(auth, node, **kwargs):
 
     #check institutions:
     try:
-        inst = Institution.find_one(Q('email_domain', 'eq', auth.user.username.split('@')[1]))
+        email_domains = [email.split('@')[1] for email in auth.user.emails]
+        inst = Institution.find_one(Q('email_domain', 'in', email_domains))
```

Comment:
Apparently this works. Have done (minor) testing, and Seto said he has mongo docs that support it.